packages/medusa/src/services/totals.ts

```diff
@@ -260,21 +271,33 @@ class TotalsService extends TransactionBaseService {
    * @param opts - options
    * @return the calculated subtotal
    */
-  getSubtotal(cartOrOrder: Cart | Order, opts: SubtotalOptions = {}): number {
+  async getSubtotal(
+    cartOrOrder: Cart | Order,
+    opts: SubtotalOptions = {}
+  ): Promise<number> {
     let subtotal = 0
     if (!cartOrOrder.items) {
       return subtotal
     }
 
-    cartOrOrder.items.map((item) => {
+    const getLineItemSubtotal = async (item: LineItem): Promise<number> => {
+      const totals = await this.getLineItemTotals(item, cartOrOrder, {
+        include_tax: true,
+        exclude_gift_cards: true,
```

Comment:
question: this seams to be right as the subtotal never includes the gift card. But at the same time I want to be sure that there is no side effects.

Basically, when we includes the gift card there is an infinity loop as it require the sub total as well